src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.cs

```diff
@@ -472,24 +472,22 @@ private RuntimeTypeBuilder(string szName, int genParamPos, RuntimeTypeBuilder de
         }
 
         internal RuntimeTypeBuilder(
-            string fullname, TypeAttributes attr, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type? parent, Type[]? interfaces, RuntimeModuleBuilder module,
+            string name, TypeAttributes attr, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type? parent, Type[]? interfaces, RuntimeModuleBuilder module,
             PackingSize iPackingSize, int iTypeSize, RuntimeTypeBuilder? enclosingType)
         {
-            ArgumentException.ThrowIfNullOrEmpty(fullname);
+            if (name[0] == '\0')
```

Comment:
Could you please move these checks from the abstract base type to concrete implementations for consistency:

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/AssemblyBuilder.cs#L20-L23

https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/TypeBuilder.cs#L66-L67